### PR TITLE
add option --out-dir to packge vector_graphics_compiler

### PR DESCRIPTION
**Pull request by FursanAbdulhak**
add new option --out-dir to specific output directory when specific --input-dir option

### CODE REVIEW

File: packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart [hunk 1]
```diff
@@ -144,7 +149,23 @@ Future<void> main(List<String> args) async {
...
+      if (results.wasParsed('out-dir')) {
+        final Directory outDir = Directory(results['out-dir'] as String);
+        //to add the output dirctory if it exist
+        if (!outDir.existsSync()) {
+          outDir.createSync();
+        }
+        if (Platform.isWindows) {
+          outputPath = '${outDir.path}\\${file.path.split("\\").last}.vec';
+        } else {
+          outputPath = '${outDir.path}/${file.path.split("/").last}.vec';
```
Comment (dnfield) [inline packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:165]: Use package:path's join method instead.

**Review by dnfield — APPROVED** [review]
LGTM, thanks!

@jonahwilliams fyi

**Comment by dnfield** [comment]
(assuming tests pass haha)

File: packages/vector_graphics_compiler/pubspec.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ dependencies:
   path_parsing: ^1.0.1
   xml: ^6.3.0
   vector_graphics_codec: 1.1.8
+  path: ^1.8.3
```
Comment (jonahwilliams) [inline packages/vector_graphics_compiler/pubspec.yaml:18]: Doesn't this add a dependency on path to all users of flutter_svg?

Comment (dnfield) [reply packages/vector_graphics_compiler/pubspec.yaml:18]: ahhh I think we cheated on this before and put it in dev_dependencies...

Comment (dnfield) [reply packages/vector_graphics_compiler/pubspec.yaml:18]: We could also use a more relaxed constraint?

Comment (jonahwilliams) [reply packages/vector_graphics_compiler/pubspec.yaml:18]: I think you can cheat if its only used in the bin directory, might want to double check. I'd be fine with that.

Comment (jonahwilliams) [reply packages/vector_graphics_compiler/pubspec.yaml:18]: If that doesn't work, I'd be fine with a loose constraint like we have

Comment (dnfield) [reply packages/vector_graphics_compiler/pubspec.yaml]: I changed this to ^1.8.0. I'll fix up the analysis issues unrelated to this patch separately.

**Pull request merged**